tests/apps/search/test_indexers_subjects.py

```diff
@@ -115,3 +121,42 @@ def test_format_es_subject_for_api(self):
             SubjectsIndexer.format_es_subject_for_api(es_subject, "en"),
             {"id": 89, "image": "example.com/image.png", "name": "Computer science"},
         )
+
+    def test_build_es_query(self):
```

Comment:
this was not tested yet or you forgot to remove the existing test code?

Reply:
The return values for the viewsets were tested (the viewsets were previously building the queries).

Now it's tested more granularly on the indexer

Reply:
so you may want to change the test on the viewset to test what is now relevant to test? I'm thinking for example mocking the call to build_es_query and raising the exception to check that the response has the error?

Reply:
and stop testing the format of the query there... ?

Reply:
Right now the tests on the viewset methods are perhaps "less obvious" (more integration than unit) but also definitely less coupled to the internal implementation than they would be with a mock. Those cases you mention are covered by the existing tests.

I do not think it is worth it to change these tests to move testing upstream and increase coupling when we have existing tests covering what we want to cover.

Also note it would be simple enough for subjects & orgs but the courses test is 600 line long. In my opinion it would not be a good investment to rewrite them on the indexers just for the sake of it.


I suggest we keep them as they are and move them to the indexers only if/when the necessity arises

Reply:
I follow you.